server/src/test/java/org/apache/druid/server/coordinator/simulate/TestSegmentsMetadataManager.java

```diff
@@ -221,7 +242,29 @@ public List<Interval> getUnusedSegmentIntervals(
       final DateTime maxUsedStatusLastUpdatedTime
   )
   {
-    return null;
+    final List<DataSegmentPlus> sortedUnusedSegmentPluses = new ArrayList<>(unusedSegments.values());
+    sortedUnusedSegmentPluses.sort(
+        Comparator.comparingLong(
+            dataSegmentPlus -> dataSegmentPlus.getDataSegment().getInterval().getStartMillis()
+        )
+    );
+
+    final List<Interval> unusedSegmentIntervals = new ArrayList<>();
+
+    for (final DataSegmentPlus unusedSegmentPlus : sortedUnusedSegmentPluses) {
+      final DataSegment unusedSegment = unusedSegmentPlus.getDataSegment();
+      if (dataSource.equals(unusedSegment.getDataSource())) {
+        final Interval interval = unusedSegment.getInterval();
+
+        if ((minStartTime == null || interval.getStart().isAfter(minStartTime)) &&
+            interval.getEnd().isBefore(maxEndTime) &&
+            unusedSegmentPlus.getUsedStatusLastUpdatedDate().isBefore(maxUsedStatusLastUpdatedTime)) {
+          unusedSegmentIntervals.add(interval);
+        }
+      }
+    }
+    log.info("Found [%d] unused segment intervals: [%s]", unusedSegmentIntervals.size(), unusedSegmentIntervals);
```

Comment:
If this is really needed, maybe it should be a log line in `KillUnusedSegments` duty because `SqlSegmentsMetadataManager` doesn't seem to log this either.

Reply:
Removed it as the new metric should suffice